src/app/analytics/analytics.factory.js

```diff
@@ -278,540 +292,479 @@ const analyticsFactory = /* @ngInject */ function ($window, $timeout, envService
-                    revenue: purchaseTotal.toString(),
-                    shipping: undefined,
-                    tax: undefined,
-                    coupon: undefined
-                  },
-                  products: [
-                    ...cartObject
-                  ]
-                }
+    }),
+    transactionEvent: suppressErrors(function (purchaseData) {
+      // The value of whether or not user is covering credit card fees for the transaction
+      const coverFeeDecision = JSON.parse(sessionStorage.getItem('coverFeeDecision'))
+      // Parse the cart object of the last purchase
+      const transactionCart = JSON.parse(localStorage.getItem('transactionCart'))
+      // The purchaseId number from the last purchase
+      const lastTransactionId = sessionStorage.getItem('transactionId')
+      // The purchaseId number from the pruchase data being passed in
+      const currentTransactionId = purchaseData && purchaseData.rawData['purchase-number']
+      let purchaseTotal = 0
+      let totalPurchaseWithFeesTotal = 0
+      // If the lastTransactionId and the current one do not match, we need to send an analytics event for the transaction
+      if (purchaseData && lastTransactionId !== currentTransactionId) {
+        // Set the transactionId in localStorage to be the one that is passed in
+
+        sessionStorage.setItem('transactionId', currentTransactionId)
+        const cartObject = transactionCart.items.map((cartItem) => {
+          purchaseTotal += cartItem.amount
+          totalPurchaseWithFeesTotal += cartItem.amountWithFees || 0
+          return {
+            ...generateProduct(
+              cartItem,
+              {
+                price: cartItem.amount,
+                recurringDate: moment(cartItem.giftStartDate),
               }
-            })
-            // Send cover fees event if value is true
-            if (coverFeeDecision) {
-              $window.dataLayer.push({
-                event: 'ga-cover-fees-checkbox'
-              })
-            }
+            ),
+            processingFee: cartItem.amountWithFees && coverFeeDecision ? (cartItem.amountWithFees - cartItem.amount).toFixed(2) : undefined,
+            dimension1: localStorage.getItem('gaDonorType'),
+            dimension3: cartItem.frequency.toLowerCase() === 'single' ? 'one-time' : 'recurring',
+            dimension4: cartItem.frequency ? cartItem.frequency.toLowerCase() : undefined,
+            dimension6: purchaseData.paymentMeans['account-type'] ? 'bank account' : 'credit card',
+            dimension7: purchaseData.rawData['purchase-number'],
+            dimension8: 'designation',
+            dimension9: cartItem.config['campaign-code'] !== '' ? cartItem.config['campaign-code'] : undefined
+          }
+        })
+        // Send the transaction event if the dataLayer is defined
+        $window.dataLayer = $window.dataLayer || []
+        $window.dataLayer.push({
+          event: 'transaction',
```

Comment:
It looks like branded checkout is using a `purchase` event instead of a `transaction` event. Do we know if the desire is to change event names here?

Reply:
That is one question I'm going to ask William when he gets back to me on HelpScout. Maybe I'll chnage it and he can tell me to switch it back, if there is an issue.